infinite-tracing/src/main/java/com/newrelic/ResponseObserver.java

```diff
@@ -15,18 +15,21 @@
     private final Logger logger;
     private final DisconnectionHandler disconnectionHandler;
     private final AtomicBoolean shouldRecreateCall;
+    private final AtomicBoolean firstFailedPrecondition;
 
     public ResponseObserver(MetricAggregator metricAggregator, Logger logger, DisconnectionHandler disconnectionHandler,
-            AtomicBoolean shouldRecreateCall) {
+                            AtomicBoolean shouldRecreateCall, AtomicBoolean firstFailedPrecondition) {
```

Comment:
We treat the first failed condition a bit differently, immediate reconnect with no backoff.